plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt

```diff
@@ -458,6 +458,13 @@ private class DokkaDescriptorVisitor( // TODO: close this class and make it priv
         ExtraModifiers.OVERRIDE.takeIf { DescriptorUtils.isOverride(this) }
     ).toContainer()
 
+    inline fun <reified D : Documentable> ConstructorDescriptor.additionalExtras(): PropertyContainer<D> =
+        if(this.isPrimary)
+            (this as FunctionDescriptor).additionalExtras<D>() + PrimaryConstructorExtra
```

Comment:
If you are casting ConstructorDescriptor to FunctionDescriptor either way, you can refactor the whole extension into `FunctionDescriptor.additionalExtras()`